### DISCUSSION TITLE
how to generate the md5 checksum of a package ?

### DISCUSSION THREAD

**Discussion by teto** [keplerproject/luarocks#1533] [General]
I am trying to work around the lua-iconv issue mentioned at https://github.com/luarocks/luarocks-site/issues/199 so I've forked the project to modify the rockspec. Now I get
```
$ luarocks install lua-iconv-7-1.rockspec --local
Error: MD5 check for lua-iconv-7.tar.gz has failed.
```
and I am not sure how to regenerate that checksum (looked into the wiki fruitless) but I was also wondering if I could not just use another hash mechanism, more robust (since md5 is considered broken).

**Comment by hishamhm** [comment]
> I am not sure how to regenerate that checksum (looked into the wiki fruitless) 

It should be the result of running `md5sum` on the file downloaded from `source.url`.

> md5 is considered broken

We're using it not as a security measure, but just to catch mistakes made in goodwill (e.g. a maintainer bumping a rockspec version but forgetting to update the URL, or vice-versa). That field is optional.

Looks like the latest lua-iconv doesn't use it anymore, as it has switched into downloading from git:

https://github.com/lunarmodules/lua-iconv/blob/master/rockspecs/lua-iconv-7.1.0-1.rockspec

**Reply by teto** [reply]
thank you
